code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/LocalNotificationHandler.java

```diff
@@ -0,0 +1,286 @@
+        } else {
+            resumeIntent = intent;
+        }
+
+        resumeIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        resumeIntent.putExtra(NOTIFICATION_IDENTIFIER_KEY, messageID);
+        resumeIntent.putExtra(NOTIFICATION_USER_INFO_KEY, userInfo);
+
+        final int buildVersion = Build.VERSION.SDK_INT;
+        NotificationManager notificationManager =
+                (NotificationManager) appContext.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        try {
+            // if we have an activity for this notification, use it
+            PendingIntent sender;
+
+            if (buildVersion >= Build.VERSION_CODES.M) {
+                sender =
+                        PendingIntent.getActivity(
+                                appContext,
+                                senderCode,
+                                resumeIntent,
+                                PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+            } else {
+                sender =
+                        PendingIntent.getActivity(
+                                appContext,
+                                senderCode,
+                                resumeIntent,
+                                PendingIntent.FLAG_UPDATE_CURRENT);
+            }
+
+            if (sender == null) {
+                Log.debug(LOG_TAG,
+                        SELF_TAG,
+                        "Failed to retrieve sender from broadcast, unable to post notification");
+                return;
+            }
+
+            // Todo: This seems redundant as the App is first launched before handling Broadcast
+            // intent
+            // App.INSTANCE.setAppContext(context.getApplicationContext());
+            DeviceInforming systemInfoService =
+                    ServiceProvider.getInstance().getDeviceInfoService();
+            String appName = systemInfoService.getApplicationName();
+
+            // notification channels are required if api level is 26 or higher
+            if (buildVersion >= Build.VERSION_CODES.O) {
+                final NotificationChannel notificationChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID,
+                        NOTIFICATION_CHANNEL_NAME,
+                        NotificationManager.IMPORTANCE_HIGH);
+                notificationChannel.setDescription(NOTIFICATION_CHANNEL_DESCRIPTION);
+                notificationManager.createNotificationChannel((NotificationChannel) notificationChannel);
+
+                // TODO: handle setting of sound...the previous method got deprecated in API 26
+
+            }
+
+            // set all the notification properties (small icon, content title, and content text are
+            // all required)
+            // small icon shows up in the status bar
+            NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID);
```

Comment:
final NotificationCompat.Builder. nit: the builder usage looks a bit cleaner when called like:
```java
 new NotificationCompat.Builder(context, channelId)
                        .setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
                        .setPriority(Notification.PRIORITY_HIGH)
                        .setStyle(new NotificationCompat.BigTextStyle()))
                        .setSmallIcon(getSmallIcon());
```